apis/apiextensions/v1/composition_common.go

```diff
@@ -142,9 +146,41 @@ type ReadinessCheck struct {
 	// MatchInt is the value you'd like to match if you're using "MatchInt" type.
 	// +optional
 	MatchInteger int64 `json:"matchInteger,omitempty"`
+
+	// MatchCondition specifies the condition you'd like to match if you're using "MatchCondition" type.
+	// +optional
+	MatchCondition *MatchConditionReadinessCheck `json:"matchCondition,omitempty"`
+}
+
+// MatchConditionReadinessCheck is used to indicate how to tell whether a resource is ready
+// for consumption
+type MatchConditionReadinessCheck struct {
+	// Type indicates the type of condition you'd like to use.
+	// +kubebuilder:default="Ready"
+	Type string `json:"type,omitempty"`
+
+	// Status is the status of the condition you'd like to match.
+	// +kubebuilder:default="True"
+	Status string `json:"status,omitempty"`
+}
+
+// Validate checks if the match condition is logically valid.
+func (m *MatchConditionReadinessCheck) Validate() *field.Error {
+	if m == nil {
+		return nil
+	}
+	if m.Type == "" {
+		return field.Required(field.NewPath("type"), "cannot be empty for type MatchCondition")
+	}
+	if m.Status == "" {
+		return field.Required(field.NewPath("status"), "cannot be empty for type MatchCondition")
+	}
+	return nil
 }
 
 // Validate checks if the readiness check is logically valid.
+//
+//nolint:gocyclo // This function is not that complex, just a switch
```

Comment:
Slight preference for putting this after the `{` on the next line. I'm pretty sure that will stop it from showing up as parted of the generated godoc for this function.